File: examples/php/basic/basic.php
<?php
require_once __DIR__ . '/../../../mm-php/vendor/autoload.php';

use io\metamessage\mm\MetaMessage;

class Person {
    public string $name = "Ed";
    public int $age = 30;
}

// 创建 Person 对象
$person = new Person();
echo "Original: Name={$person->name}, Age={$person->age}\n";

// 编码到 Wire 格式
$wire = MetaMessage::encode($person);
echo "Encoded: " . bin2hex(implode(array_map('chr', $wire))) . "\n";

// 从 Wire 解码
$decoded = MetaMessage::decode($wire, Person::class);
echo "Decoded: Name={$decoded->name}, Age={$decoded->age}\n";

File: examples/csharp/basic/Basic.cs
using MetaMessage;

namespace Examples.Basic;

public class Person
{
    public string Name { get; set; } = "Ed";
    public int Age { get; set; } = 30;
}

public class Basic
{
    public static void Main(string[] args)
    {
        // 创建 Person 对象
        Person person = new Person();
        Console.WriteLine($"Original: Name={person.Name}, Age={person.Age}");

        // 编码到 Wire 格式
        byte[] wire = MetaMessage.MetaMessage.Encode(person);
        Console.WriteLine($"Encoded: {BitConverter.ToString(wire).Replace("-", "").ToLower()}");

        // 从 Wire 解码
        Person decoded = MetaMessage.MetaMessage.Decode<Person>(wire);
        Console.WriteLine($"Decoded: Name={decoded.Name}, Age={decoded.Age}");
    }
}
